### PR TITLE
Explain convention for organising components

### PR DESCRIPTION
**Pull request by callumlocke**
Stemming from https://github.com/Financial-Times/next-stream-page/issues/868#issuecomment-174592014

### CODE REVIEW

File: docs/developer-guide/index.md [hunk 1]
```diff
@@ -36,11 +38,13 @@ Many applications also implement some or all of:-
 - `make tidy` destroyed the dynamically spun up Heroku app.
 
 ## Conventions
+
 - We write code in U.S. English.  This is to avoid things like `color: $next-colour-blue;`.
 - We use tabs for indentation.
 - We take turns to bring biscuits into office.
+- We try to group JavaScript, CSS and template files together as ‘components’. A component should have its own appropriately named folder, and that folder should contain an `index.js` file and/or a `_main.scss` file, along with any other files specific to the component. The component folder should go inside either `server/components` or `client/components` – or, if it's used on both the server and client, `shared/components`.
```
Comment (matthew-andrews) [inline docs/developer-guide/index.md]: > and that folder should contain an

A component might only be JavaScript or Scss … so maybe say ‘ and that folder **might normally** contain an’ (?)

Comment (matthew-andrews) [inline docs/developer-guide/index.md]: oh wait you say and/or… that's fine

**Comment by matthew-andrews** [comment]
can merge??

**Comment by callumlocke** [comment]
yes, if you're happy with it

**Comment by matthew-andrews** [comment]
go go go

**Pull request merged**